From 4615a916300536c30d43c4767ad7318e9fdb83eb Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Wed, 19 Jun 2024 19:00:22 +0530
Subject: [PATCH] [WA] Fixed build error in kernelflinger.

Observed build error due to missing multiple crypto sources reference.
Adding crypto sources.

Tests: Prepared EB and there is no build error.

Tracked-On: NA
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 Android.mk               | 2 +-
 libsslsupport/Android.mk | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index 28e9b16..2558502 100644
--- a/Android.mk
+++ b/Android.mk
@@ -22,7 +22,7 @@ endif
 KERNELFLINGER_CFLAGS += -DAVB_AB_I_UNDERSTAND_LIBAVB_AB_IS_DEPRECATED
 
 ifeq ($(TARGET_UEFI_ARCH),x86_64)
-    KERNELFLINGER_CFLAGS += -D__STDC_VERSION__=199901L
+#    KERNELFLINGER_CFLAGS += -D__STDC_VERSION__=199901L
     KERNELFLINGER_CFLAGS += -DARCH_X86_64=1
 endif
 
diff --git a/libsslsupport/Android.mk b/libsslsupport/Android.mk
index cfe3afd..ed65b69 100644
--- a/libsslsupport/Android.mk
+++ b/libsslsupport/Android.mk
@@ -107,8 +107,8 @@ include $(LOCAL_PATH)/android-config.mk
 LOCAL_SRC_FILES := $(LOCAL_SRC_FILES_$(LOCAL_ARCH))
 endif
 ifneq (,$(filter boringssl, $(KERNELFLINGER_SSL_LIBRARY)))
-include $(LOCAL_PATH)/sources.mk
-LOCAL_SRC_FILES := $(crypto_sources) $(linux_$(LOCAL_ARCH)_sources)
+include $(LOCAL_PATH)/crypto-sources.mk
+LOCAL_SRC_FILES += $(crypto_sources) $(linux_$(LOCAL_ARCH)_sources)
 #ifeq ($(FIRST_BUILD_ID),O)
 LOCAL_C_INCLUDES += $(KERNELFLINGER_SSLSUPPORT_PATH)/borningssl
 LOCAL_CFLAGS += -I$(KERNELFLINGER_SSLSUPPORT_PATH)/borningssl
-- 
2.34.1

